lib/core/networking/protocol/protocol.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:convert';
+
+import 'authentication.dart';
+
+typedef Deserialize<T> = T Function(Map<String, dynamic>);
+
+class ProtocolDelegate {
+  /// A dictionary of all protocol elements and their coresponding deserializers
+  //? This should not be final because of how testing is implemented.
+  static Map<String, Deserialize> elements = <String, Deserialize>{
+    //* Authentication
+    'AuthRequest': (Map<String, dynamic> json) => AuthRequest.fromJson(json),
+    'AuthResponse': (Map<String, dynamic> json) => AuthResponse.fromJson(json),
+  };
+
+  static String serialize(dynamic message) {
+    // Get className
+    final String type = message.runtimeType.toString();
+    final String data = jsonEncode(message);
+
+    return '{"type":"$type","data":$data}';
+  }
+
+  static dynamic deSerialize(String message) {
```

Comment:
```suggestion
  static dynamic deserialize(String message) {
```
deserialize is a real word